pinot-controller/src/main/java/org/apache/pinot/controller/recommender/rules/impl/SegmentSizeRule.java

```diff
@@ -85,7 +85,12 @@ public void run()
   }
 
   /**
-   * Estimate optimal segment size parameters
+   * Estimate segment size parameters by extrapolation based on the number of records and size of the generated segment.
+   * The linear extrapolation used here is not optimal because of columnar way of storing data and usage of different
+   * indices. Another way would be to iteratively generate new segments with expected number of rows until the ideal
+   * segment is found, but that's costly because of the time it takes to generate segments. Although the extrapolation
+   * approach seems to be less accurate, it is chosen due to its performance.
```

Comment:
How long does it take? Can we do 2 or 3 iterations, maybe?

Reply:
I actually used JMH to measure the time and find a good candidate that doesn't take much time. It depends on the cardinalities and number of columns, but in average for numRows = 50K  and a 14-15 columns it takes one second. Generating an actual size segment is definitely taking more time and I don't think it's going to be a good user experience.